event-hubs.api.md

```diff
@@ -0,0 +1,287 @@
+    updateCheckpoint(sequenceNumber: number, offset: number): Promise<void>;
+    // @internal (undocumented)
+    updateCheckpoint(eventDataOrSequenceNumber: ReceivedEventData | number, offset?: number): Promise<void>;
+}
+
+// @public
+export interface PartitionContext {
+    consumerGroupName: string;
+    eventHubName: string;
+    fullyQualifiedNamespace: string;
+    partitionId: string;
+}
+
+// @public
+export interface PartitionManager {
+    claimOwnership(partitionOwnership: PartitionOwnership[]): Promise<PartitionOwnership[]>;
+    listCheckpoints(fullyQualifiedNamespace: string, eventHubName: string, consumerGroup: string): Promise<Checkpoint[]>;
+    listOwnership(fullyQualifiedNamespace: string, eventHubName: string, consumerGroupName: string): Promise<PartitionOwnership[]>;
+    updateCheckpoint(checkpoint: Checkpoint): Promise<string>;
+}
+
+// @public
+export interface PartitionOwnership extends PartitionContext {
+    eTag?: string;
+    lastModifiedTimeInMs?: number;
+    ownerId: string;
+    ownerLevel: number;
+}
+
+// @public
+export interface PartitionProperties {
+    beginningSequenceNumber: number;
+    eventHubName: string;
+    lastEnqueuedOffset: number;
+    lastEnqueuedSequenceNumber: number;
+    lastEnqueuedTimeUtc: Date;
+    partitionId: string;
+}
+
+// @public
+export type ProcessCloseHandler = (reason: CloseReason, context: SubscriptionPartitionContext) => Promise<void>;
+
+// @public
+export type ProcessErrorHandler = (error: Error, context: SubscriptionPartitionContext) => Promise<void>;
+
+// @public
+export type ProcessEvent = (receivedEvent: ReceivedEventData, context: PartitionContext & PartitionCheckpointer) => Promise<void>;
+
+// @public
+export type ProcessInitializeHandler = (context: SubscriptionPartitionContext & SubscriptionPartitionInitializer) => Promise<void>;
+
+// @public
+export interface ReceivedEventData {
+    body: any;
+    enqueuedTimeUtc: Date;
+    offset: number;
+    partitionKey: string | null;
+    properties?: {
+        [key: string]: any;
+    };
+    sequenceNumber: number;
+    systemProperties?: {
+        [key: string]: any;
+    };
+}
+
+export { RetryOptions }
+
+// @public
+export interface SendBatchOptions {
+    abortSignal?: AbortSignalLike;
+    parentSpan?: Span | SpanContext;
+}
+
+// @public
+export interface Subscription {
+    close(): Promise<void>;
+    isRunning: boolean;
+}
+
+// @public
+export interface SubscriptionEventHandlers {
+    processClose?: ProcessCloseHandler;
+    processError?: ProcessErrorHandler;
+    processEvent: ProcessEvent;
+    processInitialize?: ProcessInitializeHandler;
+}
+
+// @public
+export interface SubscriptionOptions {
+    defaultEventPosition?: EventPosition;
+    maxBatchSize?: number;
+    maxWaitTimeInSeconds?: number;
+    trackLastEnqueuedEventInfo?: boolean;
+}
+
+// @public
+export interface SubscriptionPartitionContext extends PartitionContext, PartitionCheckpointer {
```

Comment:
Same with names, why subscription? it's inferred?

Reply:
I did this one a bit late but here's the rationale:

`PartitionContext` is used in a lot of spots (and maybe it should be differently named). It's the basic informational fields - partition ID, event hub name, FQDN (sorry, namespace). This turns out to be a pretty useful set of fields and is common in quite a few spots.

For the event handlers for subscriptions we also have the lastEnqueued info. So rather than adding an optional field that's not used in other contexts I created this more specialized version of PartitionContext instead. The naming might be annoying (not wedded to it at all) but I really dislike adding fields to a shared class if it's not actually used everywhere.